pkg/reloader/reloader.go

```diff
@@ -114,6 +118,20 @@ type TriggerReloader interface {
 	TriggerReload(ctx context.Context) error
 }
 
+// CfgDirOption contains options for watching directories containing configurations. For
+// example, a directory could be contain such as additional scrape config files or rule
+// files listed in the main Prometheus configuration. Sub-directories are ignored.
+type CfgDirOption struct {
+	// Dir is the path containing the Prometheus configurations to watch.
+	Dir string
```

Comment:
Maybe something for later, but I wonder if a glob would be more flexible here? Would give people the option to select files named a certain way and organize configs.

Reply:
Created an issue to track this additional work and added a comment in the code. Thanks!